### ISSUE TITLE
Force Unwrapping rule doesn't apply to self

### ISSUE THREAD

**Issue by skagedal** [realm/SwiftLint#2759]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Linting the following snippet with `force_unwrapping` enabled in `.swiftlint.yml`: 

```swift
class Test {
    func foo() {
        DispatchQueue.main.async { [weak self] in
            var string: String?
            _ = string!
            _ = self!
        }
    }
}
```

I would expect SwiftLint to warn on both force unwrappings. However, it does not catch `self!`.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint        
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'foo.swift' (1/1)
/Users/simon/tmp/testswiftlint/foo.swift:5:23: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

0.32.0 - installed with Homebrew

```yml
opt_in_rules:
  - force_unwrapping
```

Running Xcode 10.2.1 (confirmed with `xcode-select -p`), not using nested configurations.
